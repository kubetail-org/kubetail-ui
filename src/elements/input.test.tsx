import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import { Input } from './input';

describe('Input', () => {
  it('renders contents properly', () => {
    const { asFragment } = render(<Input placeholder="Enter text" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('applies custom className', () => {
    const { getByPlaceholderText } = render(<Input placeholder="Class test" className="custom-class" />);
    expect(getByPlaceholderText('Class test')).toHaveClass('custom-class');
  });

  it('accepts and displays value', () => {
    const { getByDisplayValue } = render(<Input value="Hello" onChange={() => {}} />);
    expect(getByDisplayValue('Hello')).toBeInTheDocument();
  });

  it('renders as disabled', () => {
    const { getByPlaceholderText } = render(<Input placeholder="Disabled test" disabled />);
    expect(getByPlaceholderText('Disabled test')).toBeDisabled();
  });

  it('sets aria-invalid when invalid', () => {
    const { getByPlaceholderText } = render(<Input placeholder="Invalid test" aria-invalid />);
    expect(getByPlaceholderText('Invalid test')).toHaveAttribute('aria-invalid');
  });
});
