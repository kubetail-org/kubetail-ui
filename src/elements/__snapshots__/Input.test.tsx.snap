// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input > add a placeholder 1`] = `
<DocumentFragment>
  <input
    class="w-full rounded-md focus-visible:ring-ring shadow-sm focus-visible:outline-0 border border-border/80 file:focus:ring-ring file:h-10 file:focus:outline-none file:focus:ring-0 file:border-0 file:bg-transparent file:text-foreground disabled:opacity-50 disabled:border-border/70 disabled:cursor-not-allowed text-foreground file:text-sm file:font-medium px-3"
    placeholder="A placeholder"
  />
</DocumentFragment>
`;

exports[`Input > renders contents properly 1`] = `
<DocumentFragment>
  <input
    class="w-full rounded-md focus-visible:ring-ring shadow-sm focus-visible:outline-0 border border-border/80 file:focus:ring-ring file:h-10 file:focus:outline-none file:focus:ring-0 file:border-0 file:bg-transparent file:text-foreground disabled:opacity-50 disabled:border-border/70 disabled:cursor-not-allowed text-foreground file:text-sm file:font-medium px-3"
  />
</DocumentFragment>
`;
